storage/watchable_store.go

```diff
@@ -152,11 +164,18 @@ func (s *watchableStore) Close() error {
 	return s.store.Close()
 }
 
-func (s *watchableStore) Watcher(key []byte, prefix bool, startRev int64) (Watcher, CancelFunc) {
+func (s *watchableStore) WatchChan() WatchChan {
+	return &watchChan{
+		watchable: s,
+		ch:        make(chan storagepb.Event, chanBufLen),
+	}
+}
+
+func (s *watchableStore) watch(key []byte, prefix bool, startRev int64, ch chan storagepb.Event) (*watcher, CancelFunc) {
```

Comment:
i do not really care about the internal impl at this stage actually. but if you want, i can make it better though.